### PR TITLE
Update document, include info on DHCP and MAC

### PR DESCRIPTION
**Pull request by qubitrenegade**
- [X] Explain the feature

Related to what I found in #783, add some language to the Networking document to explain how to setup DHCP with VNET and how to cause the MAC address to be regenerated.

Let me know if you'd like me to change anything.

Thanks!
- Q

- [X ] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: doc/source/networking.rst [hunk 1]
```diff
@@ -205,6 +205,10 @@ To configure both IPv4 and IPv6:
 
 .. note:: For VNET jails, a default route has to also be specified.
 
+To enable DHCP:
+
+:samp:`# iocage set vnet=on dhcp=on bpf=yes examplejail`
+
```
Comment (sonicaj) [inline doc/source/networking.rst:215]: Actually now just setting one prop should work `dhcp=on`. This prop will automatically set the other two props to true.

Comment (qubitrenegade) [reply doc/source/networking.rst:215]: You know...  I should have tried that... :)

Ok, I'll update and repush.  Thanks! :+1:

Comment (qubitrenegade) [reply doc/source/networking.rst:215]: Neat!  This github browser editor is really cool.

Change updated as requested.  Do you need me to squash the commits into one?

Comment (skarekrow) [reply doc/source/networking.rst:215]: Actually setting `dhcp=on` is only during creation, during `set` it won't set the others as they may already have values users care about.

Comment (qubitrenegade) [reply doc/source/networking.rst:215]: ok, so just to clarify:

When creating a new clone, you can do 

```
iocage create ... dhcp=on
```

But if you're updating a container that was previously statically set, you need:

```
iocage set vnet=on dhcp=on bpf=yes
```

To ensure all of the appropriate settings are set for DHCP?

Comment (skarekrow) [reply doc/source/networking.rst:215]: When creating a new jail and cloning one yes, also correct.

Comment (qubitrenegade) [reply doc/source/networking.rst:215]: Ok, great.  I reverted the commit removing those from the networking doc.

I could add something like the following to the basic-use doc:

```
Creates a FreeBSD 11.0-RELEASE jail with the custom name *myjail* and
sets the jail to start at system boot time.

+ To create a a jail with a DHCP interface add the `dhcp=on` property:
+
+ :samp:`# iocage create -r 11.0-RELEASE --name myjail dhcp=on`
+
+ The `dhcp=on` implies creating a VNET virtual network stack and
+ enabling the Berkley Packet Filter. DHCP cannot work without a VNET.
+ More information about VNETs is available in the VNET(9) FreeBSD manual page.

More information about jail properties is available in the iocage(8)
```

Comment (skarekrow) [reply doc/source/networking.rst:215]:
```suggestion
Creates a FreeBSD 11.0-RELEASE jail with the custom name *myjail* and
sets the jail to start at system boot time.

+ To create a a jail with a DHCP interface add the `dhcp=on` property:
+
+ :samp:`# iocage create -r 11.0-RELEASE --name myjail dhcp=on`
+
+ The `dhcp=on` property implies creating a VNET virtual network stack and
+ enabling the Berkley Packet Filter. DHCP cannot work without VNET.
+ More information about VNET is available in the VNET(9) FreeBSD manual page.

More information about jail properties is available in the iocage(8)
```

(Fixed a couple typos and added a word) Sounds good to me

Comment (qubitrenegade) [reply doc/source/networking.rst:215]: Ok, I apologize for the delay here, IRL got busy and this somehow fell off my github notifications.

It's been about a month, so I'm having trouble remembering where the "Creates a FreeBSD 11.0-RELEASE jail with the custom name *myjail* and sets the jail to start at system boot time." line came from as it doesn't seem to fit in context...

Thoughts/Comments?

Thanks!

Comment (skarekrow) [reply doc/source/networking.rst:215]: You know, I was just looking this over and I'm also not sure where it grabbed that change. No apologies needed, in fact in this time a new DHCP feature was added. You can actually supply `INTERFACE|dhcp` now, so for example most users will find it useful to do `vnet0|DHCP`, this only runs DHCP on *just* the vnet0 interface as opposed to running it on all interfaces (which is what dhcp=on did)

I suppose that might not be super relevant for an easy example, thoughts?

**Review by skarekrow — APPROVED** [review]

**Comment by skarekrow** [comment]
Thanks!

**Pull request merged**